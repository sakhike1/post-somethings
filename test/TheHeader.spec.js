import { mount, createLocalVue } from "@vue/test-utils";
import TheHeader from "@/components/Navigation/TheHeader.vue";
import TheSideNavToggle from "@/components/Navigation/TheSideNavToggle";

// Create a localVue instance
const localVue = createLocalVue();

describe("TheHeader", () => {
  it('emits "sidenavToggle" event when TheSideNavToggle emits "toggle"', async () => {
    // Mount the component with localVue
    const wrapper = mount(TheHeader, {
      localVue,
      stubs: {
        TheSideNavToggle: true, // Stub the TheSideNavToggle component
        NuxtLink: true, // Stub the NuxtLink component
      },
    });

    // Trigger the toggle event in TheSideNavToggle component
    await wrapper.findComponent(TheSideNavToggle).vm.$emit("toggle");

    // Assert that the "sidenavToggle" event has been emitted by the component
    expect(wrapper.emitted().sidenavToggle).toBeTruthy();
    expect().toMatchSnapshot();
  });
});
