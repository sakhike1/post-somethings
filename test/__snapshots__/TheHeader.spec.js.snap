// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheHeader emits "sidenavToggle" event when TheSideNavToggle emits "toggle" 1`] = `undefined`;
